src/Chapter2.hs

```diff
@@ -727,7 +743,9 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate (x:xs) = replicate x x ++ smartReplicate xs
+--negative numbers will break it :)
```

Comment:
The way replicate works, it won't break it 🙂 

```haskell
>>> replicate (-1) 9
[]
```

So, it will just put an empty list instead of the elements 🙂